js/components/controlButton.js

```diff
@@ -0,0 +1,316 @@
+var Icon = require('./icon');
+var Tooltip = require('./tooltip');
+var Utils = require('./utils');
+var CONSTANTS = require('../constants/constants');
+
+/**
+ * Template component that is used for buttons that don't have accessibility enabled.
+ * The component is picked at runtime by ControlButton depending on the props that
+ * were passed to it.
+ * @param {Object} props Component's props
+ * @return {Component} React component
+ */
+var NonAccessibleButton = function(props) {
+  return (
+    <a
+      {...props}
+      aria-hidden={props.ariaHidden}>
+      {props.children}
+    </a>
+  );
+};
+
+/**
+ * Component used for action buttons within the skin. Currently used for both control
+ * bar and skip buttons. Implements common functionality for these buttons which includes
+ * icons, highlighting, tooltips and accessibility.
+ */
+var ControlButton = React.createClass({
+  /**
+   * Stores a ref to this component's main element or component.
+   * @private
+   * @param {*} buttonRef Either a reference to an AccessibleButton or a DOM element,
+   * depending on the component that was rendered
+   */
+  storeRef: function(buttonRef) {
+    this.buttonRef = buttonRef;
+    // Pass ref on to parent, if subscribed
+    if (typeof this.props.onRef === 'function') {
+      this.props.onRef(this.buttonRef);
+    }
+  },
+
+  /**
+   * Chooses a component (either AccessibleButton or NonAccessibleButton) to use
+   * as base for the ControlButton depending on whether or not accessibility is
+   * enabled for this particular instance. Accessibility is enabled by default
+   * unless the ariaHidden prop is set to false explicitly.
+   * @private
+   * @return {Component} An AccessibleButton or NonAccessibleButton component, depending on the case
+   */
+  getButtonComponent: function() {
+    var Component;
+
+    if (this.props.ariaHidden === true) {
+      Component = NonAccessibleButton;
+    } else {
+      Component = AccessibleButton;
+    }
+    return Component;
+  },
+
+  /**
+   * Extracts the default styles for the button icon from the skin.json.
+   * @private
+   * @return {Object} An object with color and opacity properties which represent the default style of the button
+   */
+  getIconStyles: function() {
+    var iconStyles = {
+      color: Utils.getPropertyValue(this.props.skinConfig, 'controlBar.iconStyle.inactive.color'),
+      opacity: Utils.getPropertyValue(this.props.skinConfig, 'controlBar.iconStyle.inactive.opacity')
+    };
+    return iconStyles;
+  },
+
+  /**
+   * Applies the highlighted styles to the button icon on hover.
+   * TODO
+   * This was moved over from ControlBar but the original implementation sets the
+   * styles directly on the rendered element, which is an anti-pattern in React.
+   * Eventually we should update this in order to accomplish the same with setState().
+   * @private
+   * @param {Event} event The mousenter event object that triggered the action
```

Comment:
@jpfranco Is it `mousenter` or `mouseenter`? Looks like you use `onMouseEnter` as the name for the later function...

Reply:
Yes, it's with double `e`. Will fix this